### PR TITLE
Remove unnecessary non_contiguous and gradient tests from test_linalg

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #67833
* __->__ #68188
* #67996
* #68523
* #68522
* #69909
* #69908

As per title

cc @mruberry @jianyuh @nikitaved @pearu @walterddr @IvanYashchuk @xwang233 @Lezcano

Differential Revision: [D33774419](https://our.internmc.facebook.com/intern/diff/D33774419)

### CODE REVIEW

**Review by mruberry — APPROVED** [review]
Cool!

Just ping me when @IvanYashchuk and @nikitaved have gotten through this stack, too ;)

**Review by IvanYashchuk — APPROVED** [review]
> +4 −317 

The stats of this PR look really great!

File: test/test_linalg.py [hunk 1]
```diff
@@ -1124,6 +1084,7 @@ def test_eigvalsh_errors_and_warnings(self, device, dtype):
             with self.assertRaisesRegex(RuntimeError, "tensors to be on the same device"):
                 torch.linalg.eigvalsh(t, out=out)
 
+<<<<<<< HEAD
```
Comment (IvanYashchuk) [inline test/test_linalg.py:1087]: Git conflict markers have leaked into the PR.

Comment (Lezcano) [reply test/test_linalg.py:1087]: Oooops. Thanks for the catch, fixed now.

**Comment by mruberry** [comment]
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33774419).

**Comment by mruberry** [comment]
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33774419).

**Comment by mruberry** [comment]
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33774419).

**Comment by mruberry** [comment]
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33774419).

**Pull request closed without merging**